drivers/azure/azureutil/azureutil.go

```diff
@@ -467,6 +471,31 @@ func (a AzureClient) CreateVirtualMachine(resourceGroup, name, location, size, a
 	log.Debugf("OS disk blob will be placed at: %s", osDiskBlobURL)
 	log.Debugf("SSH key will be placed at: %s", sshKeyPath)
 
+	var osProfile *compute.OSProfile
+	if os == drivers.LINUX {
+		osProfile = &compute.OSProfile{
+			ComputerName:  to.StringPtr(name),
+			AdminUsername: to.StringPtr(username),
+			LinuxConfiguration: &compute.LinuxConfiguration{
+				DisablePasswordAuthentication: to.BoolPtr(true),
+				SSH: &compute.SSHConfiguration{
+					PublicKeys: &[]compute.SSHPublicKey{
+						{
+							Path:    to.StringPtr(sshKeyPath),
+							KeyData: to.StringPtr(sshPublicKey),
+						},
+					},
+				},
+			},
+		}
+	} else if os == drivers.WINDOWS {
+		osProfile = &compute.OSProfile{
+			ComputerName:         to.StringPtr(name),
+			AdminUsername:        to.StringPtr(username),
+			AdminPassword:        to.StringPtr(password),
+			WindowsConfiguration: &compute.WindowsConfiguration{},
+		}
+	}
```

Comment:
if you could handle `else` as well (by erroring out) that would be great. I feel like we should have an osutil file where we host the windows/linux distinction codes. Otherwise inline if/else OS-handling code will get longer and longer.